### ISSUE TITLE
View Or Check Network Policy Dropped Traffic

### ISSUE THREAD

**Issue by mhobotpplnet** [projectcalico/calico#1923]
Where can I check logs for dropped traffic by Network Policy utilizing Calico? 

For example: ingress is denied for all namespaces to namespaceA. 
Where would one be able to check the logs that say namespaceB tried to do a connection to namespaceA however it was dropped due to Network Policy? 

Or check dropped connections from external source ? -- I tried finding it in various log places including nodes, no-go thus far. 

Am trying to troubleshoot one incident in which network policy was applied to namespaceA and now the external source in private network is not able to talk to namespaceA. I tried adding cidr for that private network however the packets are still dropped regardless which cidr I enable -- unless its like 0.0.0.0/0 :)

**Comment by mhobotpplnet** [comment]
Just to note, calico-felix is showing debug level= INFO .. so am not 100% sure if thats why I am not seeing all logs I should be like "ERROR" WARN etc.

**Comment by tmjd** [comment]
Packets are not dropped by calico-felix, calico writes iptables rules and then iptables is responsible for allowing or dropping traffic. You can write NetworkPolicy that will log out when the policy is hit, see https://docs.projectcalico.org/v3.1/reference/calicoctl/resources/networkpolicy#rule and use the `Log` action.

One way to troubleshoot why some traffic is being dropped is to simply checkout the packet counts in iptables. You can do `sudo iptables -Z` to clear the packet counts, attempt the traffic, and then `sudo iptables-save -c | grep DROP` to see what dropped the traffic (you can leave off the grep and see packet counts for much more and that can help too).

**Comment by mhobotpplnet** [comment]
Thank you @tmjd appreciate the answer. I do use k8 network policy  I assume that I will need to use clico network policy utilizing the calicoctl ?

**Comment by mhobotpplnet** [comment]
I tried what was suggested above, have really not gotten much info from node it self..

```
[0:0] -A cali-tw-calicec66c -m comment --comment "cali:FzmLU-sOh4" -m comment --comment "Drop if no profiles matched" -j DROP
[0:0] -A cali-tw-cali3be4 -m comment --comment "cali:baCW-TFNv" -m conntrack --ctstate INVALID -j DROP
[0:0] -A cali-tw-calif134 -m comment --comment "cali:fLewC1con" -m comment --comment "Drop if no policies passed packet" -m mark --mark 0x0/0x2000000 -j DROP
[0:0] -A cali-tw-calif13b8e4 -m comment --comment "cali:c92kVFZ0" -m comment --comment "Drop if no profiles matched" -j DROP
```

**Comment by tmjd** [comment]
Yes you would have to write policy using Calico policy (loaded with calicoctl).  Kubernetes policies are given an order of 1000 so if you create a logging policy with order greater than 1000 that matches all traffic I think you should see all dropped packets logged in the iptables logs.

Though if you're not seeing any packets being dropped then creating a policy won't help.  Your original post was that you were trying to log dropped traffic from an external source, is that still the case?
How are you accessing the service? Are you maybe using a Kubernetes Service/NodePort, if so you would have to make sure you are checking for dropped traffic on the node with the destination pod.
